### ISSUE TITLE
Extend image summary so it allows displaying a label next to an image

### ISSUE THREAD

**Issue by panmari** [tensorflow/tensorflow#335]
It would be really neat if the tensorboard could additionally show the predicted and/or given label for an image in the 'images' tab.

**Comment by girving** [comment]
@danmane: Thoughts?

**Comment by bschreck** [comment]
This seems pretty important. If labels come from a separate file and need to be joined with the proper images in tensor flow, then having a good and easy visualization would do wonders for being confident that the correct labels are matched with each image.

**Comment by danmane** [comment]
From an API perspective, how do you imagine this working? The ImageSummary op would take in an image input and also (optionally?) a text label input? I've also heard requests for bounding boxes on the images. (What happens if you want to make multiple predictions on a single image?)

The image viewer is definitely one of the least-developed parts of TensorBoard at the moment. I want to improve it in the future, but it's not very high on my queue at the moment as I'm prioritizing features that are domain independent and thus useful to every user of TensorBoard (general usability, hyperparameter search, etc). If you want this to get fixed sooner, the best way is to come up with a plan for what the API should look like, get buy-in, and then submit a pull request - I'd be happy to help coordinate the process :)

**Comment by danmane** [comment]
Once we have plugin support in TensorBoard, this will be a good candidate for a plugin. But I'm going to close it for now since it won't be easy to write plugins for a few months.

**Comment by golmschenk** [comment]
It appears TensorBoard supports plugins now. Will this issue be reopened soon?

**Comment by MrWanter** [comment]
+ 1 for plot detection label, not only bounding boxes

**Comment by MrWanter** [comment]
+1 for plot detection label, not only boxes

**Comment by shianiawhite** [comment]
For anyone still interested in this issue, now that TensorBoard has a separate repository from TensorFlow, I have reopened the issue there. [Extend image summary so it allows displaying a label next to an image](https://github.com/tensorflow/tensorboard/issues/2204)
